netket/optimizer/stochastic_reconfiguration.py

```diff
@@ -0,0 +1,303 @@
+import numpy as _np
+from scipy.linalg import lstsq as _lstsq
+from scipy.sparse.linalg import LinearOperator
+from netket.stats import sum_on_nodes as _sum_on_nodes
+from scipy.sparse.linalg import cg, gmres, minres
+from mpi4py import MPI
+
+
+class SR:
+    r"""
+    Performs stochastic reconfiguration (SR) updates.
+    """
+
+    def __init__(
+        self,
+        lsq_solver=None,
+        diag_shift=0.01,
+        use_iterative=True,
+        is_holomorphic=True,
+        svd_threshold=None,
+        sparse_tol=None,
+        sparse_maxiter=None,
+    ):
+
+        self._lsq_solver = lsq_solver
+        self._diag_shift = diag_shift
+        self._use_iterative = use_iterative
```

Comment:
No need to set it here, as it is changed below anyway.

Reply:
Indeed it is not changed below unless lsq_solver is not None